src/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.RunAsClient.cs

```diff
@@ -0,0 +1,91 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Security.Principal;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.IO.Pipes.Tests
+{
+    public class NamedPipeTest_RunAsClient : RemoteExecutorTestBase
+    {
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public async Task RunAsClient_Windows()
+        {
+            string pipeName = Path.GetRandomFileName();
+            using (var server = new NamedPipeServerStream(pipeName))
+            using (var client = new NamedPipeClientStream(".", pipeName, PipeDirection.InOut, PipeOptions.None, TokenImpersonationLevel.Impersonation))
+            {
+                Task serverTask = server.WaitForConnectionAsync();
+
+                client.Connect();
+                await serverTask;
+
+                server.RunAsClient(() => { });
+            }
+        }
+
+        [DllImport("libc", SetLastError = true)]
+        internal static extern unsafe int seteuid(uint euid);
+
+        [DllImport("libc", SetLastError = true)]
+        internal static extern unsafe uint geteuid();
+
+        public static bool IsSuperUser()
+        {
+            return geteuid() == 0;
+        }
+
+        [ConditionalFact(nameof(IsSuperUser))]
+        [PlatformSpecific(TestPlatforms.AnyUnix)]
+        public void RunAsClient_Unix()
+        {
+            string pipeName = Path.GetRandomFileName();
+            uint pairID = (uint)(Math.Abs(new Random(5125123).Next()));
+            RemoteInvoke(ServerConnectAsId, pipeName, pairID.ToString()).Dispose();
+        }
+
+        public class GetEuidTestClass
+        {
+            public uint pairID;
+            public void AssertEqual()
+            {
+                Assert.Equal(pairID, geteuid());
+            }
+        }
+
+        private static int ServerConnectAsId(string pipeName, string pairIDString)
+        {
+            uint pairID = uint.Parse(pairIDString);
+            Assert.NotEqual(-1, seteuid(pairID));
```

Comment:
I assumed you could only use seteuid with valid user IDs, but if you're generating them randomly and if seteuid is successful every time, obviously that's not the case.